csrc/tensor_view.cpp

```diff
@@ -1168,15 +1168,36 @@ TensorView* TensorView::cacheFork() {
 TensorView* TensorView::cacheAfter(
     LoadStoreOpType op_type,
     CacheOp cache_op,
-    bool propagate_allocation_domain) {
+    bool propagate_allocation_domain,
+    std::vector<Expr*> cached_uses) {
   NVF_ERROR(
       !container()->isA<kir::Kernel>(),
       "Function invalid for kernel container.");
   FusionGuard fg(fusion());
 
+  if (!cached_uses.empty()) {
+    std::unordered_set<Expr*> unique_uses = fusion()->unordered_uses(this);
+    for (auto use : cached_uses) {
+      NVF_ERROR(
+          unique_uses.count(use),
+          "cached_uses is not among the use of the TensorView");
+    }
+  } else {
+    // avoid non-determinism and ensure unique
+    std::unordered_set<Expr*> unique_uses;
+    auto this_uses = uses();
```

Comment:
Is it possible to have duplicates?

Reply:
I don't think so. since we have `uses_` as private and addUse does skip duplicates https://github.com/NVIDIA/Fuser/blob/a5022da00e2e429228e78b309fb78bc3da3cf49c/csrc/ir/base_nodes.cpp#L96-L102

But I'm a bit scared by just leaving it not checked, since it's only a `std::vector`.

Reply:
If so, let's make it an error if a duplicate is found.

Reply:
switched. I'll kick off the CI to see if there's any surprises. :crossed_fingers: